zerver/views/upload.py

```diff
@@ -69,9 +69,12 @@ def upload_file_backend(request, user_profile):
         return json_error(_("You may only upload one file at a time"))
 
     user_file = list(request.FILES.values())[0]
-    if ((settings.MAX_FILE_UPLOAD_SIZE * 1024 * 1024) < user_file._get_size()):
+    file_size = user_file._get_size()
+    if ((settings.MAX_FILE_UPLOAD_SIZE * 1024 * 1024) < file_size):
         return json_error(_("Uploaded file is larger than the allowed limit of %s MB") % (
             settings.MAX_FILE_UPLOAD_SIZE))
+    if (not check_upload_quota(user_profile, file_size)):
+        return json_error(_("Your upload quota can't be exceeded"))
```

Comment:
I think we should write this as "Upload would exceed your maximum quota."; and maybe add a second sentence "Consider deleting some previously uploaded files." to give them an actionable thing they can do to fix the situation.